ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1036,248 @@ def test_server_option_with_unreachable_ad(self):
             self.master.run_command(['rm', '-f', ad_zone_file])
             tasks.configure_dns_for_trust(self.master, self.ad)
             self.remove_trust(self.ad)
+
+
+class TestNonPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with non posix AD trust
+    """
+    topology = 'line'
+    num_ad_domains = 1
+    num_clients = 1
+
+    def test_add_nonposix_trust(self):
+        self.add_trust('ipa-ad-trust')
+
+    def test_modify_auto_private_groups_value_default_trusted_range(self):
+        """
+        Modify existing range for default trusted AD domain range
+        with auto-private-groups set as true
+        """
+        domainusers = 'domain users'
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group("true")
+        result = self.clients[0].run_command(["id", nonposixuser])
+        uid_reg = (
+            r"^uid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        gid_reg = (
+            r"gid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        grp_reg = (
+            r"groups=(\d+)\({0}\),(\d+)\({1}@{2}\)".format
+            (nonposixuser, domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
```

Comment:
The test is also missing a test case: with nonposixuser1 (a non posix user with a different primaryGroup), ensure that the user is returned with uid different from gid (the primaryGroup is used for gid).